### PR TITLE
Integration tests: Add notifications and improve user runtime updates

### PR DESCRIPTION
**Pull request by N-o-X**
This PR adds tests for notifications including relations (user, usergroup and recipient).

Runtime update tests for users didn't really update anything until now, but only created new ones. This has been changed and now updates the same object over and over for testing.

### CODE REVIEW

**Comment by julianbrost** [comment]
The change outside of `tests/` should probably get its own commit with an explanation.

**Comment by N-o-X** [comment]
> The change outside of `tests/` should probably get its own commit with an explanation.

Removed them and will open a separate PR for that.

File: tests/object_sync_test.go [hunk 1]
```diff
@@ -403,33 +447,166 @@ func TestObjectSync(t *testing.T) {
...
+				})
+
+				require.Eventuallyf(t, func() bool {
+					var count int
+					err := db.Get(&count, "SELECT COUNT(*) FROM notification WHERE name = ?", baseNotification.fullName())
+					require.NoError(t, err, "querying notification count should not fail")
+					return count == 1
+				}, 20*time.Second, 200*time.Millisecond, "notification with name=%q should exist in database", baseNotification.fullName())
+
+				// Currently broken, but has been tested manually multiple times. Gets more time after RC2
```
Comment (julianbrost) [inline tests/object_sync_test.go:533]: Probably add a `TODO:` here.

File: tests/object_sync_test.go [hunk 2]
```diff
@@ -710,6 +887,168 @@ func makeTestUsers(t *testing.T) []User {
...
+		return *n.HostName + "!" + n.Name
+	} else {
+		return *n.HostName + "!" + *n.ServiceName + "!" + n.Name
+	}
+}
+
+func (n Notification) verify(t require.TestingT, db *sqlx.DB) {
+	verifyIcingaDbRow(t, db, n)
+
+	{
```
Comment (julianbrost) [inline tests/object_sync_test.go:919]: I'd add a comment before each of the following three block that states the table it checks.

File: tests/object_sync_test.go [hunk 3]
```diff
@@ -710,6 +887,168 @@ func makeTestUsers(t *testing.T) []User {
...
+		err := db.Select(&rows, query, n.fullName())
+
+		require.NoError(t, err, "mysql query")
+		require.Equal(t, expected, rows, "Recipients in database should be equal")
+	}
+
+	// User
+	// Groups
+	// Recipients
+	// Times{Begin,End}
```
Comment (julianbrost) [inline tests/object_sync_test.go:1006]: Looks like a leftover to me. Either remove or add a proper TODO comment

**Review by julianbrost — CHANGES_REQUESTED** [review]

File: tests/object_sync_test.go (hunk 2 above)
Comment (N-o-X) [reply tests/object_sync_test.go:921]: Done

File: tests/object_sync_test.go [hunk 4]
```diff
@@ -1059,4 +1399,4 @@ func makeCustomVarTestData(t *testing.T) []*CustomVarTestData {
 	})
 
 	return data
-}
+}
```
Comment (julianbrost) [inline tests/object_sync_test.go:1402]: Please add that newline again :)

**Review by julianbrost — CHANGES_REQUESTED** [review]

**Review by julianbrost — APPROVED** [review]
Looks good, but please only merge after https://github.com/Icinga/icinga2/pull/9009 was merged and tests are passing here.

**Pull request merged**